.github/workflows/workflow.yml

```
@@ -72,8 +72,23 @@ jobs:
     name: codecov
     runs-on: ubuntu-latest
     steps:
-    - uses: codecov/codecov-action@v4.5.0
-      with:
-        fail_ci_if_error: true # optional (default = false)
-        token: ${{ secrets.CODECOV_TOKEN }} # required
-        verbose: true # optional (default = false)
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v4
+      - name: Generate coverage report
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install pytest
+          python -m pip install pytest-cov
+          python -m pip install .[dev]
+          python -m pytest --cov=./ --cov-report=xml:pytest_cov.xml
```

Can this be 2 actions maybe?

```suggestion
      - name: Install dependencies
        run: python -m pip install .[dev]
      - name: Generate coverage report
        run: python -m pytest --cov=./ --cov-report=xml:pytest_cov.xml
```

(I'm guessing that `.[dev]` already has `pytest` and `pytest-cov` but please check)

---

Can you add `python -m pip install --upgrade pip` before installing `.[dev]` please

---

Applied suggestion with the addition of upgrading `pip` in [2ddc2af](https://github.com/poliastro/czml3/pull/131/commits/2ddc2af82ff5c9b965b29c8c7c6e8eaa66282ecf)